docs/site/migration/components/overview.md

```diff
@@ -0,0 +1,85 @@
+---
+lang: en
+title: 'Migrating components and extensions'
+keywords: LoopBack 4, LoopBack 3, Migration, Extensions, Components
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/migration-extensions-overview.html
+redirect_from: /doc/en/lb4/migration-extensions.html
+---
+
+{% include tip.html content="
+Missing instructions for your LoopBack 3 use case? Please report a [Migration docs issue](https://github.com/strongloop/loopback-next/issues/new?labels=question,Migration,Docs&template=Migration_docs.md) on GitHub to let us know.
+" %}
+
+{% include note.html content="
+This is a placeholder page, the task of adding content is tracked by the
+following GitHub issue:
+[loopback-next#3955](https://github.com/strongloop/loopback-next/issues/3955)
+" %}
+
+<!--
+FIXME: Create a new follow-up task to create a diagram per Raymond's comment:
+> Maybe a diagram would help if it shows the handshake between an application
+> and a component as well as typical artifacts exported from a component.
+-->
+
+In LoopBack 3, a component is essentially a function that extends and/or patches
+the target application.
+
+LoopBack 3 components can contribute:
+
+- additional models
+- new REST API endpoints (Express routes)
+- mixin to be used by application models
+
+They can also modify application's models to
+
+- register new
+  [Operation Hooks](https://loopback.io/doc/en/lb3/Operation-hooks.html)
+- register new [Remote Hooks](https://loopback.io/doc/en/lb3/Remote-hooks.html)
+- define new relations
+
+In LoopBack 4, a component is typically a class providing artifacts it wants to
+contribute to the application. It is responsibility of the target application to
+import those artifacts.
+
+LoopBack 4 components can contribute:
+
+<!-- FIXME change items to links pointing to doc pages -->
+
+- Model and Entity classes
+- Mixins
+- Sequence Actions
+- Controllers
+- Life cycle observers
+- Repositories
+- Service proxies
+- Servers
+- Extensions for Extension Points
+  - Booters
+  - Model API builders
+  - Extensions for Extension Points defined by 3rd-party component
+- Any other values to be bound in `Application`'s context:
+  - Classes
+  - Providers
+  - Arbitrary bindings (instances of `Binding` class)
+
+As the last resort, LoopBack 4 components can also modify the target application
+directly by calling `Application` APIs (this is similar to LoopBack 3 approach).
+
+To make the migration guide easier to navigate, we split component-related
```

Comment:
We should mention https://github.com/strongloop/loopback-next/pull/5304 and give an example to show how to use an application as a component to allow `boot` artifacts from a sub-application.

Reply:
I am afraid I don't see how the booter for loading artifacts from a sub-application is related to migrating LB3 extensions to LB4? Can you @raymondfeng please open a follow-up issue to describe in more detail what are you looking for?